e2e/cypress/integration/channel_sidebar/dm_gm_behaviour_spec.js

```diff
@@ -0,0 +1,257 @@
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod
+// Group: @dm_category
+
+import {getAdminAccount} from '../../support/env';
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+describe('MM-T3156 DM category', () => {
+    const sysadmin = getAdminAccount();
+    let testUser;
+    before(() => {
+        // # Enable channel sidebar organization
+        cy.apiUpdateConfig({
+            ServiceSettings: {
+                ExperimentalChannelSidebarOrganization: 'default_on',
+            },
+        });
+
+        // # Login as test user and visit town-square
+        cy.apiInitSetup({loginAfter: true}).then(({team, user}) => {
+            testUser = user;
+            cy.visit(`/${team.name}/channels/town-square`);
+
+            // # upgrade user to sys admin role
+            cy.externalRequest({user: sysadmin, method: 'put', path: `users/${user.id}/roles`, data: {roles: 'system_user system_admin'}});
+        });
+    });
+
+    it('MM-T2016_1 Opening a new DM should make sure the DM appears in the sidebar', () => {
+        // # Create a new user to start a DM with
+        cy.apiCreateUser().then(({user}) => {
+            // * Verify that we can see the sidebar
+            cy.get('#headerTeamName').should('be.visible');
+
+            // # Click the + button next to the DM category
+            cy.get('.SidebarChannelGroupHeader_addButton').should('be.visible').click();
+
+            // # Search for the new user's username
+            cy.get('#selectItems input').
+                type(user.username, {force: true}).
+                wait(TIMEOUTS.HALF_SEC);
+
+            // # Select the user you searched for
+            cy.get(`#displayedUserName${user.username}`).click().wait(TIMEOUTS.HALF_SEC);
+
+            // # Click Go
+            cy.findByTestId('saveSetting').should('be.visible').click();
+
+            // * Verify that a DM channel shows up in the sidebar
+            cy.get(`.SidebarLink:contains(${user.username})`).should('be.visible');
+        });
+    });
+
+    it('MM-T2016_2 Opening a new DM with a bot should make sure the DM appears in the sidebar, and the bot icon should be present', () => {
+        // # Create a new user to start a DM with
+        cy.apiCreateBot().then(({bot}) => {
+            // * Verify that we can see the sidebar
+            cy.get('#headerTeamName').should('be.visible');
+
+            // * Verify that a DM channel shows up in the sidebar
+            cy.get(`.SidebarLink:contains(${bot.username})`).should('be.visible');
+
+            // * Verify bot icon appears
+            cy.get(`.SidebarLink:contains(${bot.username})`).within(() => {
+                cy.get('i.icon-robot-happy').should('be.visible');
+            });
+        });
+    });
+
+    it('MM-T2016_3_4 Receiving a DM from a user should show the DM in the sidebar', () => {
```

Comment:
I've updated the test case and combined steps 3 and 4.
```suggestion
    it('MM-T2016_3 Receiving a DM from a user should show the DM in the sidebar', () => {
```